x-pack/plugins/maps/public/components/layer_addpanel/view.js

```diff
@@ -25,19 +25,31 @@ export class AddLayerPanel extends Component {
 
   state = {
     sourceType: null,
+    isLoading: false,
+    hasLayerSelected: false,
+    layer: null
+  }
+
+  static getDerivedStateFromProps(props) {
```

Comment:
This is just copying values from the props to the state. Just remove this, and remove `isLoading` and `hasLayerSelected` from the state.